src/adera/Machines/UserControl.cpp

```diff
@@ -129,42 +107,79 @@ void SysMachineUserControl::update_sensor(ActiveScene &rScene)
     for (ActiveEnt ent : view)
     {
         MachineUserControl &machine = view.get<MachineUserControl>(ent);
-        auto& throttlePos = std::get<wiretype::Percent>(machine.m_woThrottle.value()).m_value;
 
-        float throttleRate = 0.5f;
-        auto delta = throttleRate * rScene.get_time_delta_fixed();
+        // Get the Percent Panel which contains the Throttle Port
+        auto *panelPercent = rScene.reg_try_get< ACompWirePanel<Percent> >(ent);
 
-        if (!machine.m_enable)
+        if (panelPercent != nullptr)
         {
-            continue;
+            WirePort<Percent> const *portThrottle
+                    = panelPercent->port(MachineUserControl::smc_woThrottle);
+
+            if (portThrottle != nullptr)
+            {
+                // Get the connected node and its value
+                WireNode<Percent> &nodeThrottle = rNodesPercent.get_node(portThrottle->m_nodeIndex);
+                float throttlePos = nodeThrottle.m_state.m_value.m_percent;
+
+                float throttleRate = 0.5f;
+                auto delta = throttleRate * rScene.get_time_delta_fixed();
+
+                if (usrCtrl.m_throttleMore.trigger_hold())
+                {
+                    throttlePos = std::clamp(throttlePos + delta, 0.0f, 1.0f);
+                }
+
+                if (usrCtrl.m_throttleLess.trigger_hold())
+                {
+                    throttlePos = std::clamp(throttlePos - delta, 0.0f, 1.0f);
+                }
+
+                if (usrCtrl.m_throttleMin.triggered())
+                {
+                    SPDLOG_LOGGER_TRACE(rScene.get_application().get_logger(),
+                                      "Minimum throttle");
+                    throttlePos = 0.0f;
+                }
+
+                if (usrCtrl.m_throttleMax.triggered())
+                {
+                    SPDLOG_LOGGER_TRACE(rScene.get_application().get_logger(),
+                                      "Maximum throttle");
+                    throttlePos = 1.0f;
+                }
+
+                // Write possibly new throttle value to node
+                SysSignal<Percent>::signal_assign(
+                        rScene, {throttlePos}, nodeThrottle,
+                        portThrottle->m_nodeIndex, updPercent);
+            }
         }
 
-        if (usrCtrl.m_throttleMore.trigger_hold())
-        {
-            throttlePos = std::clamp(throttlePos + delta, 0.0f, 1.0f);
-        }
-
-        if (usrCtrl.m_throttleLess.trigger_hold())
-        {
-            throttlePos = std::clamp(throttlePos - delta, 0.0f, 1.0f);
-        }
+        // Get the Attitude Control Panel which contains the Throttle Port
+        auto *panelAttCtrl = rScene.reg_try_get< ACompWirePanel<AttitudeControl> >(ent);
```

Comment:
Not a huge fan of the `fnc< AnotherTemplatedType<T> >()` formatting to be honest, kind of screws with my mental template disambiguator. I agree that template args are difficult to read when you have a ton of nesting but I don't think this makes it any easier to read; if it's really a problem then do
```cpp
SomeTemplateType<
    std::pair<A, B>,
    std::vector<T>,
    RandomType
   > myVariable
```
but inserting spaces around the angle brackets just makes them easy to mistake for comparison operators

Reply:
I vaguely remember them causing compile errors before. They end up becoming bitshifts in some cases.

Reply:
That shouldn't be happening in any situation where it's ambiguous, if it's ambiguous it will require the `template` keyword. Was there a specific place that happened?